package.json

```diff
@@ -25,67 +31,61 @@
     "react": "^16.13.1 || ^17.0.0 || ^18.0.0 || ^19.0.0"
   },
   "devDependencies": {
-    "@babel/core": "^7.10.5",
-    "@babel/preset-react": "^7.10.4",
     "@ckeditor/ckeditor5-dev-bump-year": "^38.0.0",
     "@ckeditor/ckeditor5-dev-ci": "^38.0.0",
     "@ckeditor/ckeditor5-dev-release-tools": "^38.0.0",
     "@ckeditor/ckeditor5-dev-utils": "^38.0.0",
-    "@testing-library/react-hooks": "^8.0.1",
-    "@types/react": "^16.14.35",
-    "@types/react-dom": "^16.9.18",
-    "babel-loader": "^8.1.0",
-    "babel-plugin-istanbul": "^6.1.0",
-    "chai": "^4.2.0",
-    "ckeditor5": "alpha",
+    "@testing-library/dom": "^10.3.1",
+    "@testing-library/react": "^16.0.0",
+    "@types/react": "^18.0.0",
+    "@types/react-dom": "^18.0.0",
+    "@vitejs/plugin-react-swc": "^3.7.0",
+    "@vitest/browser": "^2.0.0",
+    "@vitest/coverage-istanbul": "^2.0.0",
+    "@vitest/ui": "^2.0.0",
+    "ckeditor5": "^42.0.0",
     "coveralls": "^3.1.1",
-    "enzyme": "^3.11.0",
-    "enzyme-adapter-react-16": "^1.15.2",
     "eslint": "^7.19.0",
     "eslint-config-ckeditor5": "^5.3.2",
     "eslint-plugin-react": "^7.32.2",
     "husky": "^8.0.2",
-    "karma": "^6.3.17",
-    "karma-chai": "^0.1.0",
-    "karma-chrome-launcher": "^3.1.0",
-    "karma-coverage": "^2.0.3",
-    "karma-mocha": "^2.0.1",
-    "karma-mocha-reporter": "^2.2.5",
-    "karma-sinon": "^1.0.5",
-    "karma-sourcemap-loader": "^0.3.7",
-    "karma-webpack": "^5.0.0",
     "lint-staged": "^10.2.11",
     "listr2": "^6.5.0",
     "minimist": "^1.2.5",
-    "mocha": "^10.0.0",
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1",
-    "sinon": "^14.0.0",
-    "terser-webpack-plugin": "^4.2.3",
-    "ts-loader": "^9.4.2",
-    "typescript": "^4.9.5",
-    "webpack": "^5.77.0",
-    "webpack-cli": "^4.9.2"
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
+    "react16": "npm:react@^16.0.0",
+    "react16-dom": "npm:react-dom@^16.0.0",
+    "react18": "npm:react@^18.0.0",
+    "react18-dom": "npm:react-dom@^18.0.0",
+    "react19": "npm:react@19.0.0-beta-26f2496093-20240514",
+    "react19-dom": "npm:react-dom@19.0.0-beta-26f2496093-20240514",
+    "typescript": "^5.0.0",
+    "vite": "^5.3.1",
+    "vitest": "^2.0.0",
+    "webdriverio": "^8.39.0"
   },
   "resolutions": {
-    "strip-ansi-cjs": "1.0.0",
-    "string-width": "^4.0.0",
+    "wrap-ansi": "7.0.0",
+    "string-width": "4.1.0",
     "semver": "^7.0.0"
   },
   "engines": {
     "node": ">=18.0.0"
   },
   "scripts": {
+    "dev": "echo \"Use 'dev:16', 'dev:18', or 'dev:19' depending on the version of React you want to test\"",
+    "dev:16": "REACT_VERSION=16 vite",
+    "dev:18": "REACT_VERSION=18 vite",
+    "dev:19": "REACT_VERSION=19 vite",
+    "build": "vite build && tsc --emitDeclarationOnly",
+    "test": "vitest run --coverage",
```

Comment:
As I understand it correctly, the tests will be processed with React 18 by default. Should we provide similar scripts for other React versions and one main command that will process tests using the all main supported React versions?

Reply:
@DawidKossowski I tried to do that, but there is versioning issue. Latest React Testing Library requires minimum React 18, and does not work on 16. On the other hand, older RTL does not support React 19.

Reply:
We've tried that, but it's more difficult than running the development environment with different versions of React.

Different versions of React have slightly different APIs for starting it. While in demos we only call these APIs in two places where we can use simple `if` statements, in tests we do a lot more of it.
Also, in the development environment, React is the only thing we need to swap. But in tests, it's also different versions of the React testing library, which makes it even more tricky.

It's not impossible, but it would add a lot of complexity.